[Release] v7.0.0: Upgrade RN to 0.80.3 and apply new architecture

mariasenosiain opened the pull request:
## Version 7.0.0
- Upgrade React Native version to 0.80.3 and apply new architecture.
- Upgrade example app/tooling
- Upgrade Android SDK version to 4.3.0: edge-to-edge compatibility + 16Kb page-size support:
        - https://github.com/getyoti/yoti-doc-scan-android/tree/v4.2.0
        - https://github.com/getyoti/yoti-doc-scan-android/releases/tag/v4.2.1 
        - https://github.com/getyoti/yoti-doc-scan-android/releases/tag/v4.3.0
        
Followed RN Upgrade helper instructions to migrate from 0.70.0 to 0.78.3: https://react-native-community.github.io/upgrade-helper/?from=0.70.0&to=0.78.3&package=com.example&name=example and then to 0.78.3 to 0.80.3 https://react-native-community.github.io/upgrade-helper/?from=0.78.3&to=0.80.3

B&T:
- [ ] Android SDK works.
- [ ] iOS SDK works.

mariasenosiain committed:
Android - Upgrade RN to 0.78.3 and apply new architecture

- Followed RN Upgrade helper instructions to migrate from 0.70.0 to 0.78.3: https://react-native-community.github.io/upgrade-helper/?from=0.70.0&to=0.78.3&package=com.example&name=example

Related to YD-6861

mariasenosiain committed:
iOS - Upgrade RN to 0.78.3 and apply new architecture

Followed RN Upgrade helper instructions to migrate from 0.70.0 to 0.78.3: https://react-native-community.github.io/upgrade-helper/?from=0.70.0&to=0.78.3&package=com.example&name=example

Related to YD-6861

mariasenosiain committed:
Edge to Edge support for React view (Android)

Related to YD-6861

mariasenosiain committed:
Set version to 6.0.0

Related to YD-6861

mircea-yoti commented on example/android/app/src/main/java/com/example/MainApplication.java:
```diff
@@ -1,66 +0,0 @@
-  public void onCreate() {
-    super.onCreate();
-    SoLoader.init(this, false);
-    initializeFlipper(this);
```
Flipper isn't needed anymore? Are there any other related code/dependencies we should remove? 🤔

mariasenosiain replied:
I don't think there are other dependencies we need to remove 👀

mircea-yoti commented on example/android/app/build.gradle:
```diff
@@ -1,40 +1,89 @@
+/**
+ * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
+ */
+def enableProguardInReleaseBuilds = false
```
This was previously `true`, are we disabling it on purpose or is it just copied from the migration helper?

mariasenosiain replied:
Ops! Yeah, I copied it from the migration helper and forgot about it 😅

mariasenosiain replied:
Sorted here https://github.com/getyoti/yoti-doc-scan-react-native/pull/50/commits/2ac0bc2b46e85425acf1382e5afc38b1a2230968

mircea-yoti commented on android/build.gradle:
```diff
@@ -7,16 +7,13 @@ def safeExtGet(prop, fallback) {
-        }
+        minSdkVersion safeExtGet('minSdkVersion', 24)
+        targetSdkVersion safeExtGet('targetSdkVersion', 35)
+        versionCode 603
```
This clashes with the version code used for `RN 6.0.0-alpha.3`: https://github.com/getyoti/yoti-doc-scan-react-native/pull/51/changes#diff-197b190e4a3512994d2cebed8aff5479ff88e136b8cc7a4b148ec9c3945bd65aR15

mariasenosiain replied:
Sorted with master merge https://github.com/getyoti/yoti-doc-scan-react-native/pull/50/commits/97696a87dc7ca21217293b340a04a75ace50d7b6

mariasenosiain replied:
And changed version number here https://github.com/getyoti/yoti-doc-scan-react-native/pull/50/commits/2ac0bc2b46e85425acf1382e5afc38b1a2230968

mariasenosiain committed:
Merge branch 'master' into YD-6861

mariasenosiain commented on android/build.gradle:
```diff
@@ -7,16 +7,13 @@ def safeExtGet(prop, fallback) {
-        }
+        minSdkVersion safeExtGet('minSdkVersion', 24)
+        targetSdkVersion safeExtGet('targetSdkVersion', 35)
+        versionCode 603
```
Sorted here https://github.com/getyoti/yoti-doc-scan-react-native/pull/50/commits/97696a87dc7ca21217293b340a04a75ace50d7b6

mariasenosiain commented on example/ios/Demo.xcodeproj/project.pbxproj:
```diff
@@ -99,8 +129,7 @@
 					13B07F861A680F5B00A75B9A = {
-						LastSwiftMigration = 1120;
-						ProvisioningStyle = Automatic;
+						LastSwiftMigration = 2620;
```
This is sorted by the previous release

mariasenosiain commented on example/src/App.tsx:
```diff
@@ -0,0 +1,180 @@
+    <SafeAreaProvider>
+      <SafeAreaView style={styles.container}>
+        <Header />
+        <View style={{flex: 1, justifyContent: 'space-between', width: '100%'}}>
```
I would prefer to not to touch the sample app react code, this is what was configrued in App.js, I just copied and pasted it

mariasenosiain commented on example/src/App.tsx:
```diff
@@ -0,0 +1,180 @@
+}
+
+function Result({code, description}: ResultProps) {
+  const isSuccessOutcome = code == 0;
```
Same about the App.tsx vs old App.js: just copied the content and I would prefer to not to touch it in this PR and keep the focus in native updates and RN version upgrade

mariasenosiain committed:
Misc:

- Set 7.0.0 as release version
- Back to iOS 13.0 as min supported version
- Enable proguard for release builds
- Set AGP and Kotlin fixed versions

Related to YD-6861

asklausen commented on package.json:
```diff
@@ -21,3 +21,4 @@
         "yoti-doc-scan-react-native.podspec"
     ]
 }
+
```
```suggestion
```

asklausen commented on android/build.gradle:
```diff
@@ -7,13 +7,13 @@ def safeExtGet(prop, fallback) {
-        versionName "6.0.0"
+        minSdkVersion safeExtGet('minSdkVersion', 24)
+        targetSdkVersion safeExtGet('targetSdkVersion', 35)
+        versionCode 605
```
```suggestion
        versionCode 700
```

mariasenosiain committed:
Upgrade to RN  0.80.3

Related with YD-6861

mariasenosiain committed:
Update package.json

Co-authored-by: anders <Anders.skovk@gmail.com>

mariasenosiain committed:
Update android/build.gradle

Co-authored-by: anders <Anders.skovk@gmail.com>

mariasenosiain committed:
Upgrade Android SDK version to 4.3.0

Related to YD-6861
